src/SarifCli/Rules/SarifValidationSkimmerBase.cs

```diff
@@ -0,0 +1,331 @@
+            {
+                Notification[] toolNotifications = run.ToolNotifications.ToArray();
+                string toolNotificationsPointer = runPointer.AtProperty(SarifPropertyName.ToolNotifications);
+
+                for (int iToolNotification = 0; iToolNotification < toolNotifications.Length; ++iToolNotification)
+                {
+                    Notification toolNotification = toolNotifications[iToolNotification];
+                    string toolNotificationPointer = toolNotificationsPointer.AtIndex(iToolNotification);
+
+                    Visit(toolNotification, toolNotificationPointer);
+                }
+            }
+
+            if (run.ConfigurationNotifications != null)
+            {
+                Notification[] configurationNotifications = run.ConfigurationNotifications.ToArray();
+                string configurationNotificationsPointer = runPointer.AtProperty(SarifPropertyName.ConfigurationNotifications);
+
+                for (int iConfigurationNotification = 0; iConfigurationNotification < configurationNotifications.Length; ++iConfigurationNotification)
+                {
+                    Notification configurationNotification = configurationNotifications[iConfigurationNotification];
+                    string configurationNotificationPointer = configurationNotificationsPointer.AtIndex(iConfigurationNotification);
+
+                    Visit(configurationNotification, configurationNotificationPointer);
+                }
+            }
+        }
+
+        private void Visit(Stack stack, string stackPointer)
+        {
+            if (stack.Frames != null)
+            {
+                StackFrame[] frames = stack.Frames.ToArray();
+                string framesPointer = stackPointer.AtProperty(SarifPropertyName.Frames);
+
+                for (int iFrame = 0; iFrame < frames.Length; ++iFrame)
+                {
+                    StackFrame frame = frames[iFrame];
+                    string framePointer = framesPointer.AtIndex(iFrame);
+
+                    Visit(frame, framePointer);
+                }
+            }
+        }
+
+        private void Visit(StackFrame frame, string framePointer)
+        {
+            Analyze(frame, framePointer);
+        }
+
+        protected virtual void Analyze(FileData fileData, string fileKey, string filePointer)
+        {
+        }
+
+        protected virtual void Analyze(PhysicalLocation physicalLocation, string physicalLocationPointer)
+        {
+        }
+
+        protected virtual void Analyze(Rule rule, string rulePointer)
+        {
+        }
+
+        protected virtual void Analyze(StackFrame frame, string framePointer)
+        {
+        }
+
+        protected void LogResult(ResultLevel level, string jPointer, string formatId, params string[] args)
+        {
+            Region region = GetRegionFromJPointer(jPointer);
+
+            // All messages start with "In {file}, at {jPointer}, ...". Prepend the jPointer to the args.
```

Comment:
Fixed the SDK to not automatically prepend the target path, upgraded SarifCli to the latest SDK, and adjusted the SarifCli resource strings accordingly.

---
In reply to: [77263861](https://github.com/Microsoft/sarif-sdk/pull/501#discussion_r77263861) [](ancestors = 77263861,77240384)